concent_api/core/views.py

```diff
@@ -472,6 +302,24 @@ def handle_send_reject_report_computed_task(request, client_message):
             requestor_public_key = client_public_key,
             status               = ReceiveStatus
         )
+        subtask = update_subtask(
+            subtask_id                  = client_message.cannot_compute_task.task_to_compute.compute_task_def['subtask_id'],
+            requestor_public_key        = client_public_key,
+            state                       = Subtask.SubtaskState.REPORTED,
+            reject_report_computed_task = client_message,
+        )
+        store_pending_message(
+            response_type       = PendingResponse.ResponseType.AckReportComputedTask,
+            client_public_key   = subtask.provider.public_key_bytes,
```

Comment:
The ack should be sent to the provider inside `ForceReportComputedTaskResponse`.

Reply:
We currently don't handle this kind of response, so I guess you are referring to `RejectReportComputedTask` ?

:+1: I wasn't aware of this, we should create separate issue for creating test for this scenario.

Reply:
Actually we return `ForceReportComputedTaskResponse`, but in database we store Ack separately. When we want to receive `Ack` to client we get it form DB and then put `Ack` inside `ForceReportComputedTaskResponse`

```python
    elif database_message.response_type == PendingResponse.ResponseType.AckReportComputedTask.name:  # pylint: disable=no-member
        if database_message.subtask.ack_report_computed_task is not None:
            ack_report_computed_task = deserialize_message(database_message.subtask.ack_report_computed_task.data.tobytes())
            mark_message_as_delivered(database_message)
            response_to_client = message.concents.ForceReportComputedTaskResponse(
                ack_report_computed_task = ack_report_computed_task,
            )
            logging.log_receive_message_from_database(
                response_to_client,
                database_message.client.public_key,
                database_message.response_type,
                database_message.queue
            )
            return response_to_client
```

Reply:
@rwrzesien No, as you see above, the code stores an ack (PendingResponse.ResponseType.AckReportComputedTask), not a reject.

@kbeker So it's against the spec. The spec does not list `AckReportComputedTask` as a valid response type to be stored in the receive queue.